libs/server-sdk/src/client_impl.cpp

```diff
@@ -116,80 +116,59 @@ ClientImpl::ClientImpl(Config config, std::string const& version)
     run_thread_ = std::move(std::thread([&]() { ioc_.run(); }));
```

Comment:
The changes made here are a stopgap to get the basic background sync + streaming use-case working (call `StartAsync`, check the future to see if the SDK is initialized.)

In the DS spec, I've tried to solve the problem of separating out the state of the system and the state of the data. They are currently intertwined. 

For example, once we have initial data, we want to say that _data is available_ (forever) - but the current data source status might become _interrupted_, so it's not a good proxy for that. 

Here when the source status changes, we ignore the new state and instead check the result of `data_system->Initialized()` to see if data is available. So a sequence of events might look like:

```
Status::kInitializing
data_system->Initialized() = false

Status::kInterrupted 
data_system->Initialized() = false

Status::kValid 
data_system->Initialized() = true

Status::kInterrupted
data_system->Initialized() = true
```


In future revisions, we should put the data state into the Status Manager so that we can support this in a first class way. There'd be no need for a `data_system->Initialized()` method because we could check the status manager instead as the source of truth.